File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
    // 1, 1, 2  1,2,3
public:
    void backtrack(vector<int>& nums, int index, vector<int>& res, vector<vector<int>>& ans){
        int n = nums.size();
        if(index>=n){
            ans.push_back(res);
            return;
        }
        res.push_back(nums[index]);
        backtrack(nums, index+1, res, ans);
        res.pop_back();
        backtrack(nums, index+1, res, ans);        
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        int n =nums.size();
        vector<int> res;
        vector<vector<int>> ans;

        backtrack(nums, 0, res, ans);
        return ans;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        my_output = [1]*len(nums)
        for i in range(len(nums)):
            for j in range(len(nums)):
                if j !=i :
                    my_output[i] = my_output[i]*nums[j]
        return my_output




        